src/phoenix/server/app.py

```diff
@@ -192,19 +193,26 @@ async def version() -> PlainTextResponse:
     return PlainTextResponse(f"{phoenix.__version__}")
 
 
+DB_MUTEX: Optional[asyncio.Lock] = None
+
+
 def _db(engine: AsyncEngine) -> Callable[[], AsyncContextManager[AsyncSession]]:
     Session = async_sessionmaker(engine, expire_on_commit=False)
 
     @contextlib.asynccontextmanager
     async def factory() -> AsyncIterator[AsyncSession]:
-        async with Session.begin() as session:
-            yield session
+        global DB_MUTEX
+        async with contextlib.AsyncExitStack() as stack:
```

Comment:
What makes the stack needed here?

Reply:
just to make the `if` statement a little easier to manage in terms of formatting